lib/src/lib/components/form-fields/amount-field/amount.helpers.ts

```diff
@@ -0,0 +1,65 @@
+import { AbstractControl, ValidationErrors, ValidatorFn } from '@angular/forms';
+import { AmountOptions } from './amount-field.model';
+
+export function getSeparatorCount(value: string, separator: string): number {
+  return value?.match(new RegExp(separator, 'g'))?.length ?? 0;
+}
+
+export function amountToNumber(
+  value: string,
+  errorCb?: (error: string) => null
+): number | null {
+  if (value?.length) {
+    const commaCount = getSeparatorCount(value, ',');
+    const pointCount = getSeparatorCount(value, '\\.');
+    // the input value can only contain one separator sign
+    if (commaCount + pointCount > 1) {
+      return errorCb ? errorCb('toManyDecimalSeparators') : null;
+    }
+    // replace comma by point to validate the string as a valid number
+    if (!pointCount && commaCount) {
+      value = value.replace(',', '.');
+    }
+    if (Number.isNaN(+value)) {
+      return errorCb ? errorCb('invalidNumber') : null;
+    }
+    return +value;
+  }
+  return null;
+}
+
+export function validateNumberInput(): ValidatorFn {
+  return (control: AbstractControl): ValidationErrors | null => {
+    let value = control.value;
+
+    if (!value?.length) {
+      return null;
+    }
+
+    let error: ValidationErrors | null = null;
+    amountToNumber(value, (e) => {
+      error = { [e]: true };
+      return null;
+    });
+    return error;
+  };
+}
+
+export function getAmountFormatter(
+  locale: string,
+  options?: AmountOptions
+): Intl.NumberFormat {
+  return new Intl.NumberFormat(locale, {
+    maximumFractionDigits: options?.maxDecimals,
+    minimumFractionDigits: options?.minDecimals,
+  });
+}
+
+export function formatAmountWithoutRounding(
+  value: number,
+  formatter: Intl.NumberFormat,
+  max: number = 0
+): string {
+  const re = new RegExp('^-?\\d+(?:.\\d{0,' + (max || -1) + '})?');
+  return formatter.format(+value.toString().match(re)[0]);
```

Comment:
You might like `Intl.NumberFormat`'s rounding mode option :)

Reply:
I looked into it, doesn't seem to work. I get the error that the option doesn't exist

Reply:
Ah no you're already using `maximumFractionDigits`—doesn't that cover the use case of this method? Not sure if I'm reading the regex right 😛

Reply:
if you type 1.456 it will round it up to 1.46 if you have 2 fractions max. For stip they don't want rounding.